editoast/src/main.rs

```diff
@@ -426,14 +426,17 @@ async fn runserver(
     let app_state = AppState::init(&args, postgres_config, redis_config).await?;
 
     // Configure the axum router
-    let router: Router<()> = axum::Router::<AppState>::new()
+    let router = axum::Router::<AppState>::new()
         .merge(views::router())
         .route_layer(axum::middleware::from_fn_with_state(
             app_state.clone(),
             authorizer_middleware,
-        ))
-        .layer(OtelInResponseLayer)
-        .layer(OtelAxumLayer::default())
+        ));
+    #[cfg(any(feature = "opentelemetry", feature = "datadog"))]
+    let router = router
+        .layer(axum_tracing_opentelemetry::middleware::OtelInResponseLayer)
```

Comment:
I recall we guessed that this layer might not be necessary since we don't need open telemetry headers in responses. Did that change?

Reply:
It seems it was not cleaned-up during the [big `axum` PR](https://github.com/OpenRailAssociation/osrd/pull/8047). I thought it was tracked in the [follow-up ticket](https://github.com/OpenRailAssociation/osrd/issues/8178), but it's not. I'm just going to do it in that PR (separate commit, see dbfcb062b).

Reply:
You're right, but I figured that since the line is in the diff, we might as well delete it now 🤷